documentation/examples/remote_storage/remote_storage_adapter/main.go

```diff
@@ -217,20 +217,8 @@ func serve(addr string, writers []writer, readers []reader) error {
 	})
 
 	http.HandleFunc("/read", func(w http.ResponseWriter, r *http.Request) {
-		compressed, err := ioutil.ReadAll(r.Body)
+		req, err := remote.DecodeRequest(r)
```

Comment:
Actually harder than I firth though, as you end up with circular imports.  Can this wait until its done it the v2 branch?  The separating out of the protos should resolve the circular imports.